server/plugin/client_test.go

```diff
@@ -617,6 +619,53 @@ func TestGetSubscriptionFilterPossibleValues(t *testing.T) {
 	}
 }
 
+func TestMakeHTTPRequest(t *testing.T) {
+	mockAPI := &plugintest.API{}
+	p := setupTestPlugin(mockAPI)
+	for _, testCase := range []struct {
+		description                       string
+		maxBytesSizeForClientResponseBody int
+	}{
+		{
+			description:                       "MakeHTTPRequest: valid",
+			maxBytesSizeForClientResponseBody: constants.MaxBytesSizeForReadingResponseBody,
+		},
+		{
+			description:                       "MakeHTTPRequest: large response body",
+			maxBytesSizeForClientResponseBody: 10000000,
```

Comment:
should we do something like this `constants.MaxBytesSizeForReadingResponseBody + 1` (or some other value) in case this constant value changes in the future.